playwright/_impl/_locator.py

```diff
@@ -0,0 +1,454 @@
+# Copyright (c) Microsoft Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pathlib
+import sys
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Awaitable,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    TypeVar,
+    Union,
+    cast,
+)
+
+from playwright._impl._api_structures import FilePayload, FloatRect, Position
+from playwright._impl._element_handle import ElementHandle
+from playwright._impl._helper import (
+    Error,
+    KeyboardModifier,
+    MouseButton,
+    locals_to_params,
+    monotonic_time,
+)
+from playwright._impl._js_handle import Serializable
+
+if sys.version_info >= (3, 8):  # pragma: no cover
+    from typing import Literal
+else:  # pragma: no cover
+    from typing_extensions import Literal
+
+if TYPE_CHECKING:  # pragma: no cover
+    from playwright._impl._frame import Frame
+    from playwright._impl._js_handle import JSHandle
+
+T = TypeVar("T")
+
+
+class Locator:
+    def __init__(self, frame: "Frame", selector: str) -> None:
+        self._frame = frame
+        self._selector = selector
+        self._loop = frame._loop
+        self._dispatcher_fiber = frame._connection._dispatcher_fiber
+
+    def __repr__(self) -> str:
+        return f"<Locator frame='{self._frame._url}' selector='{self._selector}'>"
```

Comment:
```suggestion
        return f"<Locator frame={self._frame!} selector={self._selector!r}>"
```
Better to use repr of frame